### ISSUE TITLE
Error fetching of page API when url('xxx.yyy') is present in markup code.

### ISSUE THREAD

**Issue by RafaPolit** [huridocs/uwazi#2869]
# To reproduce

Create a page with the following code:
```
<style>
  .hero {
    background-image: url('/assets/1586888468436ek11sp95rqd.png');
  }
</style>
<section class="hero">Hero</section>
```

Note that, when loading the page, within pageView there are two page API requests:
- One to the correct shared id of the page
- One to `/api/page/?sharedId=&`

This can be seen in the browser as well, there is a call to `http://localhost:3000/en/page/&`

This behavior is already present in our current production code, but that error goes 'unnoticed'.

With the introduction of #2849 , this error now tries to sets HEADERS on this error, but reply has already been sent and the instance is actually thrown with an `Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client`.

The fist fix would probably be to fix the parsing of the Markdown for this scenario, but it is also important to find out if we can prevent the instance from being thrown down if similar errors arise.

**Comment by RafaPolit** [comment]
Curiously enough, if the URL is wrapped with double quotes (") instead of single quotes ('), the double call goes away.

@daneryl any ideas?

**Comment by daneryl** [comment]
@RafaPolit 

> With the introduction of #2849 , this error now tries to sets HEADERS on this error, but reply has already been sent and the instance is actually thrown with an Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client.

this is due to the combination of lines 203 and 234

https://github.com/huridocs/uwazi/blob/305a45101ea93ba8ff7aee84739be42368d2b9b6/app/react/ServerRouter.js#L200-L235

on the 404 error the catch redirects and then the promise is rejected again, being catch again on 234 to respond the error. 
redirects already set the headers, that is why we have this error, fixed on #2870 

i think this fixes the critical part of the error, we can track the double request maybe on another issue ?
